R/merge_sce_list.R

```diff
@@ -12,41 +14,93 @@
 #'    its index in the provided `sce_list`.
 #'  - The resulting colData slot will include another new column `cell_id_column`
 #'    (default "cell_id") that will contain the SCE's original column names (i.e.
-#'    original colData rownames). Often, but not always, this rowname holds a
+#'    original colData row names). Often, but not always, this row name holds a
 #'    unique cell barcode.
+#'  - Of the original colData columns, only column names provided in the argument
+#'    `retain_coldata_cols` will be retained.
 #'  - The resulting colData rownames will be be prefixed with `{sce_name-}`.
 #'  - The resulting rowData slot column names will be appended with the given
 #'    SCE's name, as `{sce_name}-{column_name}` except for columns whose names
 #'    are indicated to preserve with the `preserve_rowdata_cols` argument.
 #'
+#'  SCE altExp contents are modified or retained as follows:
+#'  - As with the main experiment, the additional columns `batch_column` and
+#'  `cell_id_column` will be added to the colData slot.
+#'  - Of the original altExp colData columns, only column names provided in
+#'    the argument `retain_altexp_coldata_cols`, as specified for each named
+#'    altExp will be retained.
+#'  - The resulting rowData slot column names will be appended with the given
+#'    SCE's name, as `{sce_name}-{column_name}` except for the column `target_type`
+#'    which is commonly present in CITE-seq alternative experiments.
```

Comment:
I do think making `preserve_altexp_rowdata_cols` an input and a namex list makes sense. I'm not sure it is the only column that should be preserved for cellhash? I might be overthinking, but if we have things like the barcode sequence, we probably want to include those too.

Reply:
> I'm not sure it is the only column that should be preserved for cellhash? I might be overthinking

Do you mean for cite? 

Either way, here's what we have -

```
> rowData(altExp(sce_cite)) |> names()
[1] "target_type" "mean"        "detected"  

> rowData(altExp(sce_cellhash)) |> names()
[1] "barcode_id" "sample_id"  "mean"       "detected" 
```

Reply:
I did mean that. I'm surprised we don't have a `target` or `target_id` as a column? I feel like we should, even though it will be in general the same as the row name. Just me not liking important data in the row names...

Reply:
> Just me not liking important data in the row names...

Gotcha, yep. We can add this into `scpca-nf` probably here - 
https://github.com/AlexsLemonade/scpca-nf/blob/25b599adea91c28e672963577d4f10d77a90b150/bin/generate_unfiltered_sce.R#L152-L155

and then sneak into https://github.com/AlexsLemonade/scpca-nf/pull/666, if you want.